third_party/move/move-prover/bytecode-pipeline/src/memory_instrumentation.rs

```diff
@@ -59,8 +66,114 @@ impl FunctionTargetProcessor for MemoryInstrumentationProcessor {
     }
 }
 
-struct Instrumenter<'a> {
-    builder: FunctionDataBuilder<'a>,
+fn process_spec_clauses(
+    code: &[Bytecode],
+    offset: &mut usize,
+    func_env: &FunctionEnv,
+    instrumenter: &mut Instrumenter<'_>,
+) {
+    let (first_attr_id, _, exp) = match &code[*offset] {
+        Prop(attr_id, _, exp) => (attr_id, /* ignored */ (), exp),
+        _ => unreachable!("Expected a Prop"),
+    };
+
+    // Get the borrow info before the first spec clause.
+    let borrow_annotation_at = instrumenter
+        .borrow_annotation
+        .get_borrow_info_at(*offset as CodeOffset)
+        .unwrap();
+    let borrow_info = &borrow_annotation_at.before;
+
+    // Process the first clause.
+    let mut spec_instrs = vec![code[*offset].clone()];
+    let mut nodes = Vec::new();
+    collect_borrow_nodes_from_exp(&mut nodes, func_env, borrow_info, exp);
+    *offset += 1;
+
+    // Process all consecutive ones
```

Comment:
nit: processing a clause could be pulled out into its own function.

Reply:
refactored the code to check all clauses together including the first one.